ChiTech/ChiMesh/SweepUtilities/AngleSet/angleset.cc

```diff
@@ -168,13 +168,13 @@ PsiBndry(uint64_t bndry_map,
          int gs_ss_begin,
          bool surface_source_active)
 {
-  if (not surface_source_active)
-    return &ref_boundaries[bndry_map]->zero_boundary_flux[g];
-
   if (ref_boundaries[bndry_map]->IsReflecting())
     return ref_boundaries[bndry_map]->HeterogenousPsiIncoming(
       angle_num, cell_local_id, face_num, fi, gs_ss_begin);
 
+  if (not surface_source_active)
+    return &ref_boundaries[bndry_map]->zero_boundary_flux[g];
+
   return &ref_boundaries[bndry_map]->boundary_flux[g];
 }
 
```

Comment:
This is what is happening:
if reflective boundary, return A
if not surf_src_act, return B
return C (all other cases)

what when wrong was that the other was somehow incorrect. the current order makes more sense. however, the way the "logic space" is partitioned still feels weird.  do we test somewhere (input-level) that the suer cannot set a surface source on a face that has a reflective BC?

Reply:
Regarding the question ... this is essentially covered by the boundary type specification. Each boundary always has the member `zero_boundary_flux` inherited from `BoundaryBase` and all boundaries are by default mapped to a child `BoundaryVacuum` object. When the user calls the `chiLBSSetProperty` routine with the `BOUNDARY_CONDITION` option (and the reflecting option) then boundary-id XX's mapping is changed to a `BoundaryReflecting` object which will return true when `BoundaryBase::IsReflecting()` is called.

It is therefore impossible to have a surface source specified on a reflective boundary. They would be objects of different classes.

I am trying to think how to make the logic-space less weird ... But first I have to understand the cause of the weirdness weighed against the required functionality. Is it maybe because the implementation of boundaries are too far abstracted into `chi_mesh::sweep_utilities`?